sycl/include/CL/sycl/handler.hpp

```diff
@@ -1116,6 +1116,10 @@ class __SYCL_EXPORT handler {
     kernel_parallel_for_work_group<KernelName, ElementType>(KernelFunc);
   }
 
+  bool setStateExplicitKernel();
+  bool setStateSpecConstSet();
+  bool isStateExplicitKernel() const;
```

Comment:
Suggest having `void setStateExplicitKernelBundle` and `void setStateSpecConstSet` methods which do validation if the state transition is valid and throw if not.

Reply:
That does look cleaner. I have done this.